conformance-test/conformanceCommon.ts

```diff
@@ -121,29 +134,22 @@ export function executeScenario(testCase: RetryTestCase) {
           });
 
           it(`${instructionNumber}`, async () => {
+            const methodParameters: libraryMethods.ConformanceTestOptions = {
+              bucket: bucket,
+              file: file,
+              notification: notification,
+              storage: storage,
+              hmacKey: hmacKey,
+            };
+            if (testCase.preconditionProvided) {
+              methodParameters.preconditionRequired = true;
+            }
             if (testCase.expectSuccess) {
-              assert.ifError(
-                await storageMethodObject(
-                  bucket,
-                  file,
-                  notification,
-                  storage,
-                  hmacKey
-                )
-              );
+              assert.ifError(await storageMethodObject(methodParameters));
             } else {
-              try {
-                await storageMethodObject(
-                  bucket,
-                  file,
-                  notification,
-                  storage,
-                  hmacKey
-                );
-                throw Error(`${storageMethodString} was supposed to throw.`);
-              } catch (e) {
-                assert.notStrictEqual(e, undefined);
-              }
+              await assert.rejects(async () => {
```

Comment:
You do not need an async arrow function here, you can simply pass the function we are expecting to reject it does not need to be awaited.